processor/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use super::{Felt, StarkField, Vec};
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Returns the number of rows in the provided execution trace assumed to be in column-major form
+/// and contain at least one column.
+pub fn get_trace_len(trace: &[Vec<Felt>]) -> usize {
+    trace[0].len()
+}
```

Comment:
It almost feels like we should have a separate wrapper struct for `Vec<Vec<Felt>>`. We use it quite frequently and not very consistently (sometimes we allocate memory for the whole structure and then just fill in cells, in other cases we use `push` operations to grow internal vectors). This struct could have a number of useful functions, `trace_len()`, `last_row_idx()` and maybe others.

Reply:
Yeah, I agree with this. I didn't want to make changes that sweeping here, but I do think it would be an improvement. Also, at one point switching from column-major to row-major was discussed briefly. It would be nice to have everything in one struct so this would be trivial if we ever had reason to do it